pkg/catalog/routes.go

```diff
@@ -556,12 +556,22 @@ func (mc *MeshCatalog) buildInboundPolicies(t *access.TrafficTarget) []*trafficp
 
 		servicePolicy := trafficpolicy.NewInboundTrafficPolicy(buildPolicyName(destService, false), hostnames)
 
-		weightedCluster := getDefaultWeightedClusterForService(destService)
-
-		for _, sourceServiceAccount := range trafficTargetIdentitiesToSvcAccounts(t.Spec.Sources) {
-			for _, routeMatch := range routeMatches {
-				servicePolicy.AddRule(*trafficpolicy.NewRouteWeightedCluster(routeMatch, weightedCluster), sourceServiceAccount)
+		// while building inbound policies, check if the service is an apex service in traffic split
+		if mc.isApexService(destService) {
+			var localDestServices []service.MeshService
```

Comment:
Could you explain why we're checking if the service is an apex on inbound?

Reply:
In the case of building inbound routes for a particular service we need the check if its an apex service (with traffic split) cause the hostnames need to be the name of the apex service but its weighted cluster needs to be the backend service.

Reply:
We previously had it as the apex service itself but that didn't work cause we don't fully have support for multiple services on a single pod. This will require more work